test/cpp/ext/otel/otel_test_library.h

```diff
@@ -407,6 +407,18 @@ MATCHER_P7(GaugeDataIsIncrementalForSpecificMetricAndLabelSet, metric_name,
   return result;
 }
 
+MATCHER_P2(IsWithinRange, lo, hi,
+           absl::StrCat(negation ? "isn't" : "is", " between ",
+                        ::testing::PrintToString(lo), " and ",
+                        ::testing::PrintToString(hi))) {
+  return (lo) <= arg && arg <= (hi);
+}
+
+template <typename T>
+struct Extract<const IsWithinRangeMatcherP2<T, T>> {
```

Comment:
What is this for?

Reply:
we need this specialization to be able to use the above `IsWithinRange` matcher within the `HistogramResultEq` matcher.

Reply:
Please add a comment explaining that.

Reply:
done